src/TLuaInterpreter.cpp

```diff
@@ -12899,3 +12901,59 @@ Host& getHostFromLua(lua_State* L)
     assert(h);
     return *h;
 }
+
+int TLuaInterpreter::getColumnCount( lua_State * L )
+{
+    QString windowName;
+
+    if ( ! lua_gettop( L ) ) {
+        windowName = QStringLiteral("main");
+    }
+    else if ( ! lua_isstring(L, 1) ) {
+        lua_pushfstring(L, "getColumnCount: bad argument #1 type (window name as string expected, got %s)", luaL_typename(L, 1));
+        lua_error( L );
+        return 1;
+    }
+    else {
+        windowName = QString::fromUtf8(lua_tostring(L, 1));
+    }
+
+    int columns;
+    Host * pHost = &getHostFromLua(L);
+
+    if ( windowName.isEmpty() || ! windowName.compare(QStringLiteral("main"), Qt::CaseSensitive) )
+        columns = pHost->mpConsole->console->getColumnCount();
+    else
+        columns = mudlet::self()->getColumnCount(pHost, windowName);
+
+    lua_pushnumber( L, columns );
```

Comment:
Change this line (and the corresponding one in the next function) to:

    if (columns >= 0) {
        lua_pushnumber(L, columns);
        return 1;
    } else {
        lua_pushnil(L);
        lua_pushfstring(L, "window \"%s\" not found", windowName.toUtf8().constData());
        return 2;
    }

This is so that these functions conform to the __nil+errorMessage__ paradigm that we are slowly getting into the lua system to handle non-fatal errors at run-time.